web/developer.actyx.com/docs/how-to/app-auth/compiling-actyx.mdx

```diff
@@ -10,44 +10,85 @@ import { Tabs } from '../../../src/components'
-(base64 encoded) for the sake of consistent outputs.
-
-|         | Actyx                                           | Developer                                       |
-| ------- | ----------------------------------------------- | ----------------------------------------------- |
-| Public  | `075i62XGQJuXjv6nnLQyJzECZhF29acYvYeEOJ3kc5M8=` | `0nz1YdHu/JDmS6CImcZgOj9Y960sJOPrbZHAJO107qW0=` |
-| Private | `0WBFFicIHbivRZXAlO7tPs7rCX6s7u2OIMJ2mx9nwg0w=` | `08lUw93C+xzdxBcsYOoPVjzn8IHPJtnJW9Y/WyEu4v64=` |
+```json
+"licensing": {
+    "node": "development"
+}
+```
 
 :::
 
-To do so, you first need to generate an Ed25519 key pair, in case you do not know how to do so
-you can refer to [GitHub's tutorial](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key) on the topic.
+When running Actyx in production, you have fine grained control over which applications
+are authorized to run on which node, along with it, you can also sign applications to
+ensure the authenticity of the running application.
+
+To enable this, Actyx makes use of assymetric cryptography, more precisely
+[Ed25519](https://en.wikipedia.org/wiki/EdDSA#Ed25519) keys.
+When running your own Actyx deployment, you will need to generate these key pairs and
+compile Actyx from source.
 
-After you have your key pair, we need to download Actyx and compile it using your key.
+:::info Download Actyx's source
 
-1. Start by cloning the repository.
+To simplify the rest of the tutorial, ensure you have downloaded Actyx's source code:
 
 ```bash
 git clone git@github.com:Actyx/Actyx.git
 ```
 
-2. Navigate to `rust/actyx`.
+:::
+
+## Generate your own Actyx key pair
+
+As discussed, to roll your own Actyx deployment, you will need to generate your own key pair,
+so, without further ado, let's start.
+
+:::tip
+In upcoming tutorials, a "developer key pair" will be mentioned, the generation process
+for said key is the same as for the Actyx key (i.e. you can come back and read this section).
+:::
+
+1. Navigate to `rust/actyx`.
 
 ```bash
 cd rust/actyx
 ```
 
-3. Compile Actyx, using your public key.
+2. Using the Actyx CLI (`ax`), generate your Actyx key pair. It will ask you where to
+   store the key pair, you can use the provided directory or select your own, in either
+   case, just make sure you do not lose the key pair (you will need it).
+
+```bash
+cargo run --bin ax users keygen
+Generating public/private key pair ..
+Enter path in which to save the key (/Users/jmgd/Library/Application Support/actyx/keys/users/id):
+Your private key has been saved at /Users/jmgd/Library/Application Support/actyx/keys/users/id
+Your public key has been saved at /Users/jmgd/Library/Application Support/actyx/keys/users/id.pub
+```
+
+:::tip Example keys
+
+If you are just trying out the process and do not want to generate a key pair, you can
```

Comment:
Is there a caveat to using these example keys? Non-advanced readers might not know the implications. (e.g. me having to realize "oh right the private key is public")

Reply:
I've added a "disclaimer", warning that these keys are for demonstration purposes only